src/psyclone/tests/psyir/transformations/increase_rank_loop_arrays_test.py

```diff
@@ -0,0 +1,130 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2025, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: S. Siso, STFC Daresbury Lab
+
+''' Module containing tests for the IncreaseRankLoopArraysTrans class. '''
+
+import pytest
+from psyclone.psyir.nodes import Loop
+from psyclone.psyir.transformations import IncreaseRankLoopArraysTrans
+from psyclone.transformations import TransformationError
+
+
+def test_trans_name_and_str():
+    ''' Check that the name and str method works as expected. '''
+    trans = IncreaseRankLoopArraysTrans()
+    assert trans.name == "IncreaseRankLoopArraysTrans"
+    assert str(trans) == (
+        "Increases the Rank of the supplied arrays by the iteration space of "
+        "the given loop, and update all its references")
+
+
+def test_irla_validate(fortran_reader):
+    ''' Check that the validate method works as expected. '''
+    trans = IncreaseRankLoopArraysTrans()
+
+    psyir = fortran_reader.psyir_from_source("""
+        program test
+            real, dimension(10) :: a
+            integer :: i = 3
+            a(i) = 3
```

Comment:
`validate` should always fail for this code as we won't touch the reference of `a` in this assignment?

Reply:
Please add tests for accesses before and after the target loop. Alternatively, as mentioned previously, perhaps this transformation should target a whole scoping region in which the array is local?

Reply:
I provided a different solution, let me know what you think. In this case this example should pass the validate.

Reply:
I quite like the new solution but, as commented earlier, I think it needs to be made a bit more robust.